agimus_controller/agimus_controller/ocp_base_croco.py

```diff
@@ -0,0 +1,226 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+import crocoddyl
+import mim_solvers
+import numpy as np
+import pinocchio as pin
+
+from agimus_controller.mpc_data import OCPResults, OCPDebugData
+from agimus_controller.trajectory import WeightedTrajectoryPoint
+from agimus_controller.ocp_base import OCPBase
+from agimus_controller.ocp_param_base import OCPParamsCrocoBase
+
+
+class OCPCrocoBase(OCPBase):
+    def __init__(
+        self,
+        rmodel: pin.Model,
+        cmodel: pin.GeometryModel,
+        OCPParams: OCPParamsCrocoBase,
+    ) -> None:
+        """Creates an instance of the OCPCrocoBase class. This is an example of an OCP class that inherits from OCPBase,
+        for a simple goal reaching task. The class is used to solve the Optimal Control Problem (OCP) using the Crocoddyl library.
+
+        Args:
+            rmodel (pin.Model): Robot model.
+            cmodel (pin.GeometryModel): Collision Model of the robot.
+            OCPParams (OCPParamsBase): Input data structure of the OCP.
+        """
+        self._rmodel = rmodel
+        self._cmodel = cmodel
+        self._OCPParams = OCPParams
+
+    @abstractmethod
+    @property
+    def horizon_size(self) -> int:
+        """Number of time steps in the horizon."""
+        return self._OCPParams.T
+
+    @abstractmethod
+    @property
+    def dt(self) -> np.float64:
+        """Integration step of the OCP."""
+        return self._OCPParams.dt
```

Comment:
In theory this can be obtained from `SolverBase.problem.RunningModels[0].dt`, but this makes an assumption that the fist running model is of a type `IntegratedActionModelAbstract` and all running models have the same `dt`. Other option I would suggest is to store `dt` as a class parameter and return `self._dt`

Reply:
but we can obtain it from the params class, and for clarity's sake, i believe that having many methods returning the same value may be problematic at some point